src/ui/ai/AiRound.ts

```diff
@@ -225,9 +225,10 @@ export default class AiRound implements AiRoundInterface, PromotingInterface {
   public onEngineMove = (bestmove: string) => {
     const from = <Key>bestmove.slice(0, 2)
     const to = <Key>bestmove.slice(2, 4)
+    const role = <Role>chessFormat.uciToProm(bestmove)
     this.vm.engineSearching = false
     this.chessground.apiMove(from, to)
```

Comment:
I believe you'll need to conditionally call `this.chessground.promote(to, role)` if `role` is not undefined

Reply:
Hmm. There seem to be two things going on here: the local management of the board state via Chessground, and the replay engine (https://github.com/ornicar/scalachess). Calling `addMove` sends the promo move to the replay worker, which responds with the game situation information (including FEN), which eventually gets passed to `this.chessground.set` via the `onReplayAdded` -> `apply` callbacks.

So I think you're right - in practice, you don't need to specifically call promote, as it's all handled by the replay engine and callbacks. I suppose we still have to call `chessground.apiMove` for animation purposes? although `chessground.set` also does that.